tools/check-downgrade-versions.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+#
+# This script checks that we can downgrade to all the released versions
+# less then being released. For example when releasing version 2.11.4
+# we check that we can downgrade to 2.11.3, 2.11.2, ..., 2.10.8, 2.10.7, ...
+# 2.8.2.
+#
+# The versions we can downgrade to are those mentioned in downgrade_versions
+# list in src/box/lua/upgrade.lua. For the sake of parsing simplicity
+# the versions should be between begin and end marks:
+# -- DOWNGRADE VERSIONS BEGIN
+# -- DOWNGRADE VERSIONS END
+#
+# The check is only done for the commit tagged as release version
+# (annotated tag with name like 2.11.4).
+
+set -eo pipefail
+
+error() {
+   echo "$@" 1>&2
+   exit 1
+}
+
+# Tags that should not be considered.
+tag_exceptions=`mktemp`
```

Comment:
Please remove all temporary files at exit (e.g. with `trap EXIT`).

Anyway, do you really need to use temporary files in this script? Can't you use arrays?

Reply:
Done.

```diff
--- a/tools/check-downgrade-versions.sh
+++ b/tools/check-downgrade-versions.sh
@@ -21,6 +21,12 @@ error() {
    exit 1
 }

+cleanup() {
+  rm $tag_exceptions $expected_versions $actual_versions || true
+}
+
+trap "cleanup" EXIT
+
```

Discussed f2f usage of arrays. Looks like it will complicate the script.

Reply:
You should use `rm -f`. Also, better add a helper function for creating a temporary file that would add it to the cleanup list.

Reply:
Added `-f` flag. Discussed the cleanup suggestion f2f - bash is not very suited for this.